app/src/main/java/uk/gov/android/securestore/SharedPrefsStore.kt

```diff
@@ -1,42 +1,63 @@
 package uk.gov.android.securestore
 
 import android.content.Context
+import androidx.fragment.app.FragmentActivity
 import java.security.GeneralSecurityException
 import java.security.KeyStoreException
+import kotlin.coroutines.suspendCoroutine
+import uk.gov.android.securestore.authentication.AuthenticatorPromptConfiguration
 import uk.gov.android.securestore.crypto.CryptoManager
 import uk.gov.android.securestore.crypto.RsaCryptoManager
 
 class SharedPrefsStore(
-    context: Context,
+    context: FragmentActivity,
     configuration: SecureStorageConfiguration,
-    private val cryptoManager: CryptoManager = RsaCryptoManager()
+    private val cryptoManager: CryptoManager = RsaCryptoManager(
+        context,
+        configuration.id,
+        configuration.accessControlLevel
+    )
 ) : SecureStore {
-    private val sharedPrefs = configuration.getSharedPrefs(context)
+    private val sharedPrefs = context.getSharedPreferences(configuration.id, Context.MODE_PRIVATE)
 
-    override fun upsert(key: String, value: String): String {
-        try {
-            return cryptoManager.encryptText(key, value).also { writeToPrefs(key, it) }
-        } catch (e: GeneralSecurityException) {
-            throw SecureStorageError(e)
+    override suspend fun upsert(key: String, value: String): String {
+        return suspendCoroutine { continuation ->
+            try {
+                val result = cryptoManager.encryptText(value)
+                    .also { writeToPrefs(key, it) }
+                continuation.resumeWith(Result.success(result))
+            } catch (e: GeneralSecurityException) {
+                throw SecureStorageError(e)
+            }
         }
     }
 
     override fun delete(key: String) {
         writeToPrefs(key, null)
         try {
-            cryptoManager.deleteKey(key)
+            cryptoManager.deleteKey()
         } catch (e: KeyStoreException) {
             throw SecureStorageError(e)
```

Comment:
We don't seem to be providing this exception with something contextually related to calling this function. Are we happy with doing so? I ask due to visibility purposes, in that we would need to look half way down a stacktrace before we see something that's searchable within our own code base

Reply:
Currently the spec has been written that the `delete` method will throw a `SecureStorageError` , this could be refined in the future but fits spec for now